integration/client_test.go

```diff
@@ -52,3 +52,10 @@ func TestBaseUrl(t *testing.T) {
 		assert.Equal(t, "https://another.datadoghq.com", c.GetBaseUrl())
 	})
 }
+
+func TestExtraHeader(t *testing.T) {
+	t.Run("No Extra Header for backward compatibility", func(t *testing.T) {
+		c := datadog.NewClient("foo", "bar")
+		assert.Empty(t, c.ExtraHeader)
```

Comment:
Can we also add a test that ensures an extra header was set?

Reply:
That's what I wanted to do originally but it would require some refactoring of the code as there are no interfaces, and all the methods I am interested in to test are private in the `datadog` package.

Do you think it's worth a refactoring ?

Reply:
I think we're just currently testing that the client now has a new attribute 😄 But if its refactoring the whole suite, perhaps not currently worthwhile.